packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.types.ts

```diff
@@ -481,6 +473,12 @@ export interface IContextualMenuItem {
    * @deprecated Use `text` instead.
    */
   name?: string;
+
+  /**
+   * Method to override the render of the individual menu items
+   * @default ContextualMenuItem
+   */
+  contextualMenuItemAs?: React.ComponentClass<IContextualMenuItemProps> | React.StatelessComponent<IContextualMenuItemProps>;
```

Comment:
These need a way to accept a `defaultRender` prop that can be used by an override to simply 'wrap' the original behavior instead of replacing it.

Also, most targeted overrides use `IContextualMenuItem.onRender` instead of this overall override, so perhaps we need to add a `contextualMenuItemAs` to that interface with the same conventions.

Reply:
I'm going to remove contextualMenuAs from this PR and just supply the commandBarButtonAs with the defaultRender since that is the only one which needs to be overridden